Error fetching access token

When navigating to the app's home page, the app will check if the user has recently logged in and update their `access_token`.  This check is failing with the below error:

```
TypeError: Cannot read properties of undefined (reading 'sendEmptyAccessToken')
     at /Users/jasonjones/projects/desc/packages/api/src/auth/AuthController.ts:36:25
     at Generator.next (<anonymous>)
     at /Users/jasonjones/projects/desc/packages/api/src/auth/AuthController.ts:8:71
     at new Promise (<anonymous>)
     at __awaiter (/Users/jasonjones/projects/desc/packages/api/src/auth/AuthController.ts:4:12)
     at getRefreshToken (/Users/jasonjones/projects/desc/packages/api/src/auth/AuthController.ts:77:16)
    at Layer.handle [as handle_request] (/Users/jasonjones/projects/desc/node_modules/express/lib/router/layer.js:95:5)
     at next (/Users/jasonjones/projects/desc/node_modules/express/lib/router/route.js:137:13)
     at Route.dispatch (/Users/jasonjones/projects/desc/node_modules/express/lib/router/route.js:112:3)
     at Layer.handle [as handle_request] (/Users/jasonjones/projects/desc/node_modules/express/lib/router/layer.js:95:5)
[ERROR] 10:24:12 TypeError: Cannot read properties of undefined (reading 'sendEmptyAccessToken')
```

--- a/packages/api/src/auth/AuthRouter.ts
+++ b/packages/api/src/auth/AuthRouter.ts
@@ -15,7 +15,7 @@ class AuthRouter {
             .post(passport.authenticate('local'), authController.login);
 
         AuthRouter.router.get('/logout', authController.logout);
-        AuthRouter.router.get('/refreshToken', authController.getRefreshToken);
+        AuthRouter.router.get('/refreshToken', authController.getRefreshToken.bind(authController));
     }
 }
 

--- a/packages/api/src/auth/__tests__/AuthController.test.ts
+++ b/packages/api/src/auth/__tests__/AuthController.test.ts
@@ -1,0 +1,29 @@
+import { Request, Response } from 'express';
+import authController from '../AuthController';
+
+describe('Auth Controller unit tests', () => {
+    describe('login()', () => {
+        it.todo('returns context user and access token on success');
+    });
+
+    describe('getRefreshToken()', () => {
+        it('sends empty access token if a refresh token is not provided', async () => {
+            const req = {
+                cookies: {}
+            } as Request;
+            const res = {} as Response;
+            res.json = jest.fn();
+
+            await authController.getRefreshToken(req, res);
+            expect(res.json).toHaveBeenCalledWith(
+                expect.objectContaining({
+                    success: true,
+                    message: 'new access token requested.',
+                    payload: {
+                        accessToken: ''
+                    }
+                })
+            );
+        });
+    });
+});

--- a/packages/api/src/auth/__tests__/auth.acceptance-test.ts
+++ b/packages/api/src/auth/__tests__/auth.acceptance-test.ts
@@ -103,4 +103,20 @@ describe('Auth route acceptance tests', () => {
             expect(refreshToken).toEqual('');
         });
     });
+
+    describe('GET /api/auth/refreshToken route', () => {
+        it('sends empty access token if a refresh token is not provided', async () => {
+            const response = await client.getRefeshToken();
+
+            expect(response.body).toEqual(
+                expect.objectContaining({
+                    success: true,
+                    message: 'new access token requested.',
+                    payload: {
+                        accessToken: ''
+                    }
+                })
+            );
+        });
+    });
 });

--- a/packages/api/src/clientRequest/ClientRequestController.ts
+++ b/packages/api/src/clientRequest/ClientRequestController.ts
@@ -4,7 +4,8 @@ import clientRequestService from './ClientRequestService';
 class ClientRequestController {
     createClientRequest(req: Request, res: Response): Promise<Response> {
         const { clientId, requestorId, items } = req.body;
-        return clientRequestService.createClientRequest({ clientId, requestorId, items })
+        return clientRequestService
+            .createClientRequest({ clientId, requestorId, items })
             .then((cr) => {
                 // need to remove the reference to this client request in the items since it causes a circular reference
                 // and JSON does not handle it
@@ -40,7 +41,8 @@ class ClientRequestController {
     }
 
     getAllClientRequests(_: Request, res: Response): Promise<Response> {
-        return clientRequestService.getAllClientRequests()
+        return clientRequestService
+            .getAllClientRequests()
             .then((requests) => {
                 return res.json({
                     success: true,
@@ -64,7 +66,8 @@ class ClientRequestController {
 
     getClientRequest(req: Request, res: Response): Promise<Response> {
         const id = req.params.id;
-        return clientRequestService.getClientRequestById(id)
+        return clientRequestService
+            .getClientRequestById(id)
             .then((cr) => {
                 if (cr) {
                     return res.json({

--- a/packages/api/src/config/passport.ts
+++ b/packages/api/src/config/passport.ts
@@ -19,7 +19,8 @@ export const passportConfig = (passport: PassportStatic): void => {
     });
 
     passport.deserializeUser((id: string, done: (err: any, user?: User) => void) => {
-        return userService.getUserById(id)
+        return userService
+            .getUserById(id)
             .then((user) => {
                 if (user) {
                     return done(null, user);

--- a/packages/api/src/item/ItemController.ts
+++ b/packages/api/src/item/ItemController.ts
@@ -31,18 +31,19 @@ class ItemController {
             }
         }
 
-        return itemService.createItem({
-            clientId,
-            category,
-            name,
-            size,
-            priority,
-            quantity,
-            status,
-            location,
-            requestorId,
-            note
-        })
+        return itemService
+            .createItem({
+                clientId,
+                category,
+                name,
+                size,
+                priority,
+                quantity,
+                status,
+                location,
+                requestorId,
+                note
+            })
             .then((item) => {
                 // need to remove the reference to the item in the note since it causes a circular reference
                 // and JSON does not handle it
@@ -71,7 +72,8 @@ class ItemController {
     }
 
     getAllItems(req: Request, res: Response): Promise<Response> {
-        return itemService.getAllItems(req.query)
+        return itemService
+            .getAllItems(req.query)
             .then((items) => {
                 let sanitizedItems;
                 if (items) {
@@ -99,7 +101,8 @@ class ItemController {
 
     getItem(req: Request, res: Response): Promise<Response> {
         const id = req.params.id;
-        return itemService.getItemById(id)
+        return itemService
+            .getItemById(id)
             .then((item) => {
                 if (item) {
                     return res.json({
@@ -130,7 +133,8 @@ class ItemController {
     updateItem(req: Request, res: Response): Promise<Response> {
         const id = req.params.id;
         const updateData = req.body;
-        return itemService.updateItem(id, updateData)
+        return itemService
+            .updateItem(id, updateData)
             .then((item) => {
                 if (item) {
                     return res.json({
@@ -160,7 +164,8 @@ class ItemController {
 
     deleteItem(req: Request, res: Response): Promise<Response> {
         const id = req.params.id;
-        return itemService.deleteItem(id)
+        return itemService
+            .deleteItem(id)
             .then((item) => {
                 if (item) {
                     return res.json({
@@ -191,7 +196,8 @@ class ItemController {
     addNoteToItem(req: Request, res: Response): Promise<Response> {
         const itemId = req.params.id;
         const { body, authorId } = req.body;
-        return itemService.addNoteToItem({ body, itemId, authorId })
+        return itemService
+            .addNoteToItem({ body, itemId, authorId })
             .then((item) => {
                 if (item) {
                     return res.json({
@@ -223,7 +229,8 @@ class ItemController {
         const itemId = req.params.id;
         const noteId = req.params.noteId;
 
-        return itemService.deleteNoteFromItem({ noteId, itemId })
+        return itemService
+            .deleteNoteFromItem({ noteId, itemId })
             .then((item) => {
                 if (item) {
                     return res.json({

--- a/packages/api/src/testUtils/TestClient.ts
+++ b/packages/api/src/testUtils/TestClient.ts
@@ -79,6 +79,12 @@ class TestClient extends BaseTestClient {
             .set('Authorization', `Bearer ${this.accessToken}`);
     }
 
+    public getRefeshToken(token?: string): Test {
+        return request(this.app)
+            .get('/api/auth/refreshtoken')
+            .set('Cookie', [`qid=${token}`]);
+    }
+
     public loginUser(email: string, password: string): Test {
         return request(this.app)
             .post('/api/auth/login')

--- a/packages/api/src/user/UserController.ts
+++ b/packages/api/src/user/UserController.ts
@@ -11,7 +11,8 @@ class UserController {
         // only includes the accepted properties
         const { firstName, lastName, email, password, program } = req.body;
 
-        return userService.createUser({ firstName, lastName, email, password, program })
+        return userService
+            .createUser({ firstName, lastName, email, password, program })
             .then(async (user) => {
                 const baseUrl = (req.get('origin') as string) || '';
                 await mailer.sendVerificationEmail(baseUrl, user);
@@ -37,7 +38,8 @@ class UserController {
     }
 
     getAllUsers(_: Request, res: Response): Promise<Response> {
-        return userService.getAllUsers()
+        return userService
+            .getAllUsers()
             .then((users) => {
                 return res.json({
                     success: true,
@@ -59,7 +61,8 @@ class UserController {
 
     getUser(req: Request, res: Response): Promise<Response> {
         const id = req.params.id;
-        return userService.getUserById(id)
+        return userService
+            .getUserById(id)
             .then((user) => {
                 if (user) {
                     return res.json({
@@ -94,7 +97,8 @@ class UserController {
         // only includes the accepted properties
         const data = req.body as UpdatableUserFields;
 
-        return userService.updateUser(id, data)
+        return userService
+            .updateUser(id, data)
             .then((updatedUser) => {
                 if (updatedUser) {
                     return res.json({
@@ -124,7 +128,8 @@ class UserController {
 
     deleteUser(req: Request, res: Response): Promise<Response> {
         const id = req.params.id;
-        return userService.deleteUser(id)
+        return userService
+            .deleteUser(id)
             .then((deletedUser) => {
                 if (deletedUser) {
                     return res.json({
@@ -154,7 +159,8 @@ class UserController {
 
     deactivateUser(req: Request, res: Response): Promise<Response> {
         const id = req.params.id;
-        return userService.setIsActive(id, false)
+        return userService
+            .setIsActive(id, false)
             .then((deactivatedUser) => {
                 if (deactivatedUser) {
                     return res.json({
@@ -184,7 +190,8 @@ class UserController {
 
     activateUser(req: Request, res: Response): Promise<Response> {
         const id = req.params.id;
-        return userService.setIsActive(id, true)
+        return userService
+            .setIsActive(id, true)
             .then((activatedUser) => {
                 if (activatedUser) {
                     return res.json({
@@ -214,7 +221,8 @@ class UserController {
 
     confirmEmail(req: Request, res: Response): Promise<Response> {
         const emailToken = req.params.token;
-        return userService.confirmEmail(emailToken)
+        return userService
+            .confirmEmail(emailToken)
             .then((user) => {
                 if (user) {
                     return res.json({
@@ -275,7 +283,8 @@ class UserController {
     changePassword(req: Request, res: Response): Promise<Response> {
         const resetToken = req.params.token;
         const { newPassword } = req.body;
-        return userService.changePassword(resetToken, newPassword)
+        return userService
+            .changePassword(resetToken, newPassword)
             .then((user) => {
                 if (user) {
                     return res.json({

